runtime/drivers/duckdb/connectors.go

```diff
@@ -14,13 +14,24 @@ import (
 	"github.com/rilldata/rill/runtime/connectors/localfile"
 	"github.com/rilldata/rill/runtime/drivers"
 	"github.com/rilldata/rill/runtime/pkg/fileutil"
+	"go.uber.org/zap"
 )
 
 const (
 	_iteratorBatch        = 8
 	_defaultIngestTimeout = 60 * time.Minute
 )
 
+var dateTypeChangeMapping = map[string][]string{
+	"BOOLEAN":   {"BIGINT", "DOUBLE", "VARCHAR"},
+	"BIGINT":    {"DOUBLE", "VARCHAR"},
+	"DOUBLE":    {"VARCHAR"},
+	"TIME":      {"VARCHAR"},
+	"DATE":      {"TIMESTAMP", "VARCHAR"},
+	"TIMESTAMP": {"VARCHAR"},
+	"VARCHAR":   {},
```

Comment:
do we also need to add TINYINT -> INTEGER -> BIGINT as well ?

Reply:
I only added a conversion mapping of types which are inferred during auto detection. The table is mentioned here : https://duckdb.org/docs/data/csv/auto_detection#type-detection

Reply:
okay but seems like its there only for csv. For parquet such a sequence is possible.